packages/assets-controllers/src/NftController.ts

```diff
@@ -512,6 +512,53 @@ export class NftController extends BaseController<
     });
   }
 
+  /**
+   * Fetches NFT Collection Metadata from the NFT API.
+   *
+   * @param contractAddresses - The contract addresses of the NFTs.
+   * @param chainId - The chain ID of the network where the NFT is located.
+   * @returns NFT collections metadata.
+   */
+  async fetchNftCollectionsMetadata(
+    contractAddresses: string[],
+    chainId: Hex,
+  ): Promise<{
+    collections: Collection[];
+  }> {
+    return await this.#getNftCollectionInformationFromApi(
+      contractAddresses,
+      chainId,
+    );
+  }
+
+  async #getNftCollectionInformationFromApi(
+    contractAddress: string[],
+    chainId: Hex,
+  ) {
+    const urlParams = new URLSearchParams({
+      chainId,
+    });
+
+    for (const address of contractAddress) {
+      urlParams.append('contract', address);
+    }
```

Comment:
Want to mention that this API will have a limit of 20 for the contract address param in case this is relevant to your usecase

Reply:
Thanks for calling this out, this will be very rare case to have more than 20 nfts in one transaction but good to know. 
I will check later the metrics of simulations if we ever encountered this much.